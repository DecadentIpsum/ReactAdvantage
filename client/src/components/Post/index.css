.post {
  display: inline-block; }
  .post .ui-widget-content {
    border: 0 !important; }
  .post .ui-card-body {
    padding: 0 !important; }

.content {
  max-width: 300px;
  font-family: "Roboto", sans-serif; }
