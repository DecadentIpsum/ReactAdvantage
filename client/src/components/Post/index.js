import React, { Component } from 'react';
import {Card} from 'primereact/components/card/Card';

import Button from "../Button";

import "./index.css"

export default class Post extends Component {

    render() {
        let readMoreButton = <Button
            className="ui-button-secondary"
            label="Read more"
        />;

        return (
            <div className="post">
                <Card className="content" footer={readMoreButton}>
                    {this.props.text}
                </Card>

            </div>
        );
    }
}
