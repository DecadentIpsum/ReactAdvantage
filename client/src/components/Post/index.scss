.post {
    display: inline-block;

    .ui-widget-content {
        border: 0 !important;
    }

    .ui-card-body {
        padding: 0 !important;
    }
}

.grid-item-title {
    color: #1E1E1E;
    font-size: 1.5rem;
    font-family: 'Roboto', sans-serif;
    // padding: 1em;
    padding-bottom: 1rem;
    margin: 0;
    // margin-bottom: 20px;
    // margin-top: 0;
}

.content {
    max-width: 300px;
    font-family: 'Roboto', sans-serif;
}