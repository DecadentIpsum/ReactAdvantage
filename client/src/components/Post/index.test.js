import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Post from './index';

it('renders correctly', () => {
    const component = shallow(<Post />);

    expect(component).toMatchSnapshot();
});