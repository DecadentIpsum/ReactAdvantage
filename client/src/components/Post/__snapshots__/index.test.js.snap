// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Post />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        className="content"
        footer={
                <Button
                        className="ui-button-secondary"
                        label="Read more"
                />
        }
        header={null}
        id={null}
        style={null}
        subtitle={null}
        title={null}
/>,
      "className": "post",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "className": "content",
        "footer": <Button
          className="ui-button-secondary"
          label="Read more"
/>,
        "header": null,
        "id": null,
        "style": null,
        "subtitle": null,
        "title": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          className="content"
          footer={
                    <Button
                              className="ui-button-secondary"
                              label="Read more"
                    />
          }
          header={null}
          id={null}
          style={null}
          subtitle={null}
          title={null}
/>,
        "className": "post",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "className": "content",
          "footer": <Button
            className="ui-button-secondary"
            label="Read more"
/>,
          "header": null,
          "id": null,
          "style": null,
          "subtitle": null,
          "title": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
